dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESIndexUtilTest.java

```diff
@@ -0,0 +1,199 @@
+
+import com.dotcms.content.elasticsearch.util.RestHighLevelClientProvider;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.rainerhahnekamp.sneakythrow.Sneaky;
+import org.elasticsearch.ElasticsearchStatusException;
+import org.elasticsearch.action.admin.indices.settings.put.UpdateSettingsRequest;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.common.settings.Settings;
+import org.junit.BeforeClass;
+import com.dotcms.util.IntegrationTestInitService;
+import org.junit.Test;
+
+import static org.jgroups.util.Util.assertEquals;
+
+public class ESIndexUtilTest {
+
+    @BeforeClass
+    public static void prepare () throws Exception {
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+    }
+
+    /**
+     * Method to Test: {@link ESIndexUtil#isReadOnly(String...)}
+     * When: The Index is not read only
+     * Should:return false
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldReturnFalseIfTheCurrentIndicesAreReadOnly() throws DotDataException {
+        final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+
+        try {
+            putReadOnly(indiciesInfo.getWorking(), false);
+
+            final boolean readOnly = ESIndexUtil.isReadOnly(indiciesInfo.getWorking());
+
+            assertEquals(false, readOnly);
+        } finally {
+            putReadOnly(indiciesInfo.getWorking(), false);
+        }
+    }
+
+    /**
+     * Method to Test: {@link ESIndexUtil#isReadOnly(String...)}
+     * When: The Index is read only
+     * Should:return true
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldReturnTrueIfTheCurrentIndicesAreReadOnly() throws DotDataException {
+        final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+
+        try {
+            putReadOnly(indiciesInfo.getWorking(), true);
+
+            final boolean readOnly = ESIndexUtil.isReadOnly(indiciesInfo.getWorking());
+
+            assertEquals(true, readOnly);
+        } finally {
+            putReadOnly(indiciesInfo.getWorking(), false);
+        }
+    }
+
+    /**
+     * Method to Test: {@link ESIndexUtil#isReadOnly(String...)}
+     * When: The Index not exists
+     * Should: throw a {@link org.elasticsearch.ElasticsearchStatusException}
+     *
+     * @throws DotDataException
+     */
+    @Test (expected = ElasticsearchStatusException.class)
+    public void whenTheIndexNotExistsShouldThrowException() {
+        final boolean readOnly = ESIndexUtil.isReadOnly("not_Exists");
+        assertEquals(true, readOnly);
+    }
+
+    /**
+     * Method to Test: {@link ESIndexUtil#putReadOnlyToFalse(String...)}
+     * When: The index exists
+     * Should: should set it to read only
```

Comment:
this doesn't seem a correct expected result based on the method's name...

Reply:
done https://github.com/dotCMS/core/pull/18604/commits/fc979e0e1d04475951dbb213b379b17022474a8e#diff-1e54965847e19cc0a77583c4dbd18afdR87